Bottom: 299b2ebfc4379ee74e2b36335f88586ebcd3926a
Top:    c33afdcca73949284834ef4c66bbed62911b9218
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-07 01:58:44 +0000

Refresh of fix-some-strdup-calls-which-mi

---

diff --git a/src/create.c b/src/create.c
index 5262c92..4eef584 100644
--- a/src/create.c
+++ b/src/create.c
@@ -587,7 +587,7 @@ CreateNewText (LayerTypePtr Layer, FontTypePtr PCBFont,
   text->Direction = Direction;
   text->Flags = Flags;
   text->Scale = Scale;
-  text->TextString = strdup (TextString);
+  Text->TextString = (TextString && *TextString) ? strdup (TextString) : NULL;
 
   /* calculate size of the bounding box */
   SetTextBoundingBox (PCBFont, text);
